// nextra/dist/chunk-ZRPTEBDJ.mjs
// src/constants.ts
import path from "path";
var MARKDOWN_EXTENSION_REGEX = /\.mdx?$/;
var IS_PRODUCTION = process.env.NODE_ENV === "production";
var LOCALE_REGEX = /\.([a-z]{2}(-[A-Z]{2})?)$/;
var DEFAULT_LOCALE = "en-US";
var OFFICIAL_THEMES = [
  "nextra-theme-docs",
  "nextra-theme-blog"
];
var META_FILENAME = "_meta.json";
var CWD = process.cwd();
var PUBLIC_DIR = path.join(CWD, "public");
var CACHE_DIR = path.join(CWD, ".next", "cache");
var ASSET_DIR = path.join(CWD, ".next", "static", "chunks");
var EXTERNAL_URL_REGEX = /^https?:\/\//;

// src/utils.ts
import path2 from "path";
function parseFileName(filePath) {
  var _a;
  const { name, ext } = path2.parse(filePath);
  const locale = ((_a = name.match(LOCALE_REGEX)) == null ? void 0 : _a[1]) || "";
  return {
    name: locale ? name.replace(LOCALE_REGEX, "") : name,
    locale,
    ext
  };
}
var parseJsonFile = (content, path3) => {
  try {
    return JSON.parse(content);
  } catch (err) {
    console.error(`[nextra] Error parsing ${path3}, make sure it's a valid JSON`, err);
    return {};
  }
};
function truthy(value) {
  return !!value;
}
function normalizeMeta(meta) {
  return typeof meta === "string" ? { title: meta } : meta;
}

export {
  MARKDOWN_EXTENSION_REGEX,
  IS_PRODUCTION,
  DEFAULT_LOCALE,
  OFFICIAL_THEMES,
  META_FILENAME,
  CWD,
  PUBLIC_DIR,
  CACHE_DIR,
  ASSET_DIR,
  EXTERNAL_URL_REGEX,
  parseFileName,
  parseJsonFile,
  truthy,
  normalizeMeta
};
