// nextra/dist/components/index.mjs
import {
  CheckIcon,
  CopyIcon,
  WordWrapIcon
} from "../chunk-I473SOMZ.mjs";
import {
  __async,
  __objRest,
  __spreadValues
} from "../chunk-HIDP27A7.mjs";

// src/components/button.tsx
import React from "react";
var Button = (_a) => {
  var _b = _a, {
    children,
    className = ""
  } = _b, props = __objRest(_b, [
    "children",
    "className"
  ]);
  return /* @__PURE__ */ React.createElement("button", __spreadValues({
    className: [
      "nextra-button transition-colors",
      "bg-primary-700/5 border border-black/5 text-gray-600 hover:text-gray-900 rounded-md p-2",
      "dark:bg-primary-300/10 dark:border-white/10 dark:text-gray-400 dark:hover:text-gray-50",
      className
    ].join(" ")
  }, props), children);
};

// src/components/copy-to-clipboard.tsx
import React2, {
  useCallback,
  useEffect,
  useState
} from "react";
var CopyToClipboard = (_a) => {
  var _b = _a, {
    value
  } = _b, props = __objRest(_b, [
    "value"
  ]);
  const [isCopied, setCopied] = useState(false);
  useEffect(() => {
    if (!isCopied)
      return;
    const timerId = setTimeout(() => {
      setCopied(false);
    }, 2e3);
    return () => {
      clearTimeout(timerId);
    };
  }, [isCopied]);
  const handleClick = useCallback(() => __async(void 0, null, function* () {
    setCopied(true);
    if (!(navigator == null ? void 0 : navigator.clipboard)) {
      console.error("Access to clipboard rejected!");
    }
    try {
      yield navigator.clipboard.writeText(JSON.parse(value));
    } catch (e) {
      console.error("Failed to copy!");
    }
  }), [value]);
  const IconToUse = isCopied ? CheckIcon : CopyIcon;
  return /* @__PURE__ */ React2.createElement(Button, __spreadValues({
    onClick: handleClick,
    title: "Copy code"
  }, props), /* @__PURE__ */ React2.createElement(IconToUse, {
    className: "pointer-events-none h-4 w-4"
  }));
};

// src/components/code.tsx
import React3 from "react";
var Code = (_a) => {
  var _b = _a, {
    children,
    className = ""
  } = _b, props = __objRest(_b, [
    "children",
    "className"
  ]);
  const hasLineNumbers = "data-line-numbers" in props;
  return /* @__PURE__ */ React3.createElement("code", __spreadValues({
    className: [
      "border-black/5 bg-black/5 break-words rounded-md border py-0.5 px-[.25em] text-[.9em]",
      "dark:border-white/10 dark:bg-white/10",
      hasLineNumbers ? "[counter-reset:line]" : "",
      className
    ].join(" "),
    dir: "ltr"
  }, props), children);
};

// src/components/pre.tsx
import React4, { useCallback as useCallback2 } from "react";
var Pre = (_a) => {
  var _b = _a, {
    children,
    className = "",
    value,
    filename
  } = _b, props = __objRest(_b, [
    "children",
    "className",
    "value",
    "filename"
  ]);
  const toggleWordWrap = useCallback2(() => {
    const htmlDataset = document.documentElement.dataset;
    const hasWordWrap = "nextraWordWrap" in htmlDataset;
    if (hasWordWrap) {
      delete htmlDataset.nextraWordWrap;
    } else {
      htmlDataset.nextraWordWrap = "";
    }
  }, []);
  return /* @__PURE__ */ React4.createElement(React4.Fragment, null, filename && /* @__PURE__ */ React4.createElement("div", {
    className: "bg-primary-700/5 text-gray-700 absolute top-0 z-[1] w-full truncate rounded-t-xl py-2 px-4 text-xs dark:bg-primary-300/10 dark:text-gray-200"
  }, filename), /* @__PURE__ */ React4.createElement("pre", __spreadValues({
    className: [
      "bg-primary-700/5 mt-6 mb-4 overflow-x-auto rounded-xl font-medium subpixel-antialiased dark:bg-primary-300/10",
      "contrast-more:border contrast-more:border-primary-900/20 contrast-more:contrast-150 contrast-more:dark:border-primary-100/40",
      filename ? "pt-12 pb-4" : "py-4",
      className
    ].join(" ")
  }, props), children), /* @__PURE__ */ React4.createElement("div", {
    className: [
      "opacity-0 transition-opacity [div:hover>&]:opacity-100",
      "flex gap-1 absolute m-2 right-0",
      filename ? "top-8" : "top-0"
    ].join(" ")
  }, /* @__PURE__ */ React4.createElement(Button, {
    tabIndex: -1,
    onClick: toggleWordWrap,
    className: "md:hidden",
    title: "Toggle word wrap"
  }, /* @__PURE__ */ React4.createElement(WordWrapIcon, {
    className: "pointer-events-none w-4 h-4"
  })), value && /* @__PURE__ */ React4.createElement(CopyToClipboard, {
    tabIndex: -1,
    value
  })));
};

// src/components/td.tsx
import React5 from "react";
var Td = (props) => /* @__PURE__ */ React5.createElement("td", __spreadValues({
  className: "m-0 border border-gray-300 px-4 py-2 dark:border-gray-600"
}, props));

// src/components/table.tsx
import React6 from "react";
var Table = (_a) => {
  var _b = _a, {
    className = ""
  } = _b, props = __objRest(_b, [
    "className"
  ]);
  return /* @__PURE__ */ React6.createElement("table", __spreadValues({
    className: "block overflow-x-scroll " + className
  }, props));
};

// src/components/th.tsx
import React7 from "react";
var Th = (props) => /* @__PURE__ */ React7.createElement("th", __spreadValues({
  className: "m-0 border border-gray-300 px-4 py-2 dark:border-gray-600 font-semibold"
}, props));

// src/components/tr.tsx
import React8 from "react";
var Tr = (props) => /* @__PURE__ */ React8.createElement("tr", __spreadValues({
  className: "m-0 border-t border-gray-300 p-0 dark:border-gray-600 even:bg-gray-100 even:dark:bg-gray-600/20"
}, props));
export {
  Button,
  Code,
  CopyToClipboard,
  Pre,
  Table,
  Td,
  Th,
  Tr
};
